src/zc/buildout/easy_install.py

```diff
@@ -733,7 +736,20 @@ def install(self, specs, working_set=None):
                     pkg_resources.VersionConflict(dist, req), ws)
 
             best[req.key] = dist
-            extra_requirements = dist.requires(req.extras)[::-1]
+            if self._allow_unknown_extras:
+                missing_requested = sorted(
+                    set(req.extras) - set(dist.extras)
+                )
+                for missing in missing_requested:
+                    logger.warning(
+                        '%s does not provide the extra \'%s\'',
```

Comment:
It is slightly more readable if you quote it with double quotes, then you can omit the quote escapes.

Reply:
I admit I lifted that straight out of pip, quotes and all :)